controller/src/job/mod.rs

```diff
@@ -122,3 +127,116 @@ pub(crate) async fn delete_job(k8s_client: kube::Client, name: &str) -> JobResul
     let _ = result?;
     Ok(())
 }
+
+async fn get_pod(k8s_client: kube::Client, job_name: &str) -> JobResult<String> {
+    let pod_api: Api<Pod> = Api::namespaced(k8s_client, NAMESPACE);
+    let name = pod_api
+        .list(&ListParams {
+            label_selector: Some(format!("job-name={}", job_name)),
+            ..Default::default()
+        })
+        .await
+        .context(error::NotFoundSnafu {})?
+        .items
+        .first()
+        .context(error::NoPodsSnafu {
+            job: job_name.to_string(),
+        })?
+        .name_any();
+
+    Ok(name)
+}
+
+async fn pod_logs(k8s_client: kube::Client, pod_name: &str) -> JobResult<String> {
+    let log_params = LogParams {
+        follow: false,
+        pretty: true,
+        ..Default::default()
+    };
+    let pod_api: Api<Pod> = Api::namespaced(k8s_client, NAMESPACE);
+
+    pod_api
+        .logs(pod_name, &log_params)
+        .await
+        .context(error::NoLogsSnafu { pod: pod_name })
+}
+
+pub(crate) async fn archive_logs(k8s_client: kube::Client, job_name: &str) -> JobResult<()> {
```

Comment:
when does this happen? Is it only after the test has completed or failed? Or can this be kicked off at the start of the pod and the logs stream in realtime?

Reply:
It happens as a job is deleted. I wasn't able to find a solution for realtime logs.